inference/enterprise/workflows/entities/steps.py

```diff
@@ -530,23 +633,35 @@ def get_output_names(self) -> Set[str]:
 
 
 class Crop(BaseModel, StepInterface):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "description": "This block produces dynamic crops based on detections from detections-based model.",
+            "docs": "https://inference.roboflow.com/workflows/crop",
+        }
+    )
     type: Literal["Crop"]
-    name: str
-    image: Union[str, List[str]]
-    detections: str
-
-    @field_validator("image")
-    @classmethod
-    def image_must_only_hold_selectors(cls, value: Any) -> Union[str, List[str]]:
-        validate_image_is_valid_selector(value=value)
-        return value
+    name: str = Field(description="Unique name of step in workflows")
+    image: Union[InferenceImageSelector, OutputStepImageSelector] = Field(
+        description="Reference at image to be used as input for step processing",
+        examples=["$inputs.image", "$steps.cropping.crops"],
+    )
+    detections: StepOutputSelector(
```

Comment:
What’s the difference between `detections` here and `predictions` we use in a bunch of other block inputs/outputs?

Reply:
nothing, just not compliant name schema

Reply:
ok, fixed - for the time being left `detections` as auxiliary alias for that field (used secondary if two values present) - just to make the change non-breaking. Will remove alias once people would apply changes